### ISSUE TITLE
Keep shards distributed out to all nodes after reducing `replication_factor` down to 1

### ISSUE THREAD

**Issue by srfication** [qdrant/qdrant#5549]
**Is your feature request related to a problem? Please describe.**
hi qdrant team, recently i have observed that by changing the `replication_factor` in the collection meta info, the qdrant cloud will help to auto create/drop shards.

when we reduce the `replication_factor` to 1,  the shard deletion logic did not keep the shard distributed to each node causing an imbalance.

**Describe the solution you'd like**
Reducing the replication factor to 1 keep the shards distributed to all nodes rather than congregating it into 1 node. 
That is whole reason why we wanted to shard the collection.

**Describe alternatives you've considered**
Keep the shards distributed should be the default behaviour 
- or there could be a parameter alongside the replication_factor where we can indicate that the shards should be spread out to all nodes
- similar to collection creation, the shards are always spread out to all nodes

**Additional context**
Support ticket here: https://support.qdrant.io/support/tickets/5377

**Comment by timvisee** [comment]
You're right. Automatic rebalancing is still disabled at this moment. Rest assured, it should be enabled soon after our extensive testing is done.

In the mean time you can explicitly [move shards](https://qdrant.tech/documentation/guides/distributed_deployment/#moving-shards) to achieve the desired balance. Feel free to ask our support about this as they can do it for you as well.

Thank you for opening a support ticket.
